### DISCUSSION TITLE
a good ios viewer

### DISCUSSION THREAD

**Discussion by f520k510** [GerryDazoo/Slinger#126] [General]
I don't use vlc as it disconnects all the time.
I just found a player called oplayer lite.
It works quite well.

**Comment by theweatherguy** [comment]
Tried that out and it works.  (I think you mean **qplayer lite.** )   It looks like it has a long stream buffering time which it displays, which may account for its success

**Reply by flipper2222222** [reply]
https://apps.apple.com/us/app/playerxtreme-video-player/id1577168508
PlayerXtreme Video Player works well, free version plenty of features, PIP coming soon

**Reply by theweatherguy** [reply]
Thanks for that recommendation.  I'm adding both to the Mac Step By Step.

**Reply by flipper2222222** [reply]
> Thanks for that recommendation. I'm adding both to the Mac Step By Step.

you're welcome, I had to look elsewhere as VLC video and sometimes audio would freeze, and the lack of picture in picture means using the remote is very difficult, thanks to rb121 I have integrated into plex which has PIP via xteve but has an increased latency. the developers at playerexteme told me they are working on the PIP function so hopefully not too far away even if I have to pay for the premium version.

**Reply by martysm** [reply]
Is this (PlayerXtreme) still the best current information on a good place to start for a player on IOS?  MXplayer works beautifully on my Android but in trying to get my wife set up I am having zero luck with some of the MX based apps for iPhone.  I can't even find one in which I can address a network stream.  They just want to sell you services. I know nothing about the IOS ecosphere other than that I hate it.
Any advice from anyone knowledgeably will be greatly appreciated.  Thx.

**Reply by flipper2222222** [reply]
on my iphone I am still using Playerextreme , I have tried many others but no other luck yet, as stated above I am sometimes using plex
still no PIP for player extreme

**Comment by f520k510** [comment]
There is another thread on this forum about which ios video player to use.
They concluded that one can use the free vlc and change some settings.
They suggest to set deblocking filter to medium, deinterlace to on, and network cache to high latency.
It worked for me like 8 out of 10 times.
I tried oplayer and it worked every time.  I decided to pay for it to get the ad free version.

**Reply by martysm** [reply]
thanks much...I didn't see that but I'll search again..  I got vlc working but whenever it lost screen focus I'd have to quit the stream and restart it.  I made some progress with this playerextreme to where i can stream and have the remote safari page up on a split screen and hide or show the safari window whenever necessary, all on the local network.  Looking toward the next step for her it appears IOS doesn't think were grownup enough choose to use PPTP anymore.

**Reply by theweatherguy** [reply]
It's not clear if you're using a PC or a Mac as your server.   

If you're using a Mac as the Slinger server, be certain to use the preceding command before invoking the python3 command to increase the send buffer size:  

`sudo sysctl -w net.inet.tcp.sendspace=5501072`

If you’re using an M1 or M2 Mac, try this command instead of the one above: 

`sudo sysctl -w net.inet.tcp.sendspace=5601072`

Interestingly, some people have found that   `sudo sysctl -w net.inet.tcp.sendspace=5401072`   works best.  

Try all three with either Oplayer or PlayerExtreme.   

Unfortunately, Apple doesn't allow this changed value of sendspace buffer to be saved following a restart in current MacOS versions, so you need to used the command prior to every time you launch Slinger server. 

The issues playing the stream seem to all related to buffer size of the app.  But I'm sure the speed of your LAN network (and your up-speed of your internet connection if you're viewing outside of your home) plays a role.  

I've had great success with PlayerExtreme on my local network (iPad)  and Oplayer outside the home on iPhone.

**Reply by martysm** [reply]
Thanks, its on a pc at the moment.  Eventually it will be on a linux box. Then, perhaps I'll fwd the port to avoid the VPN issue, maybe not....  and I'll bite the bullet and change my VPNs to openVPN or wireshark....TBD.  Or maybe I will try to configure it all as a video source for the Plex Server.  Have to figure out the path of least resistance and greatest robustness.
The out-of-the box slinger.exe works great streaming to an old surplus samsung  (android) tablet I have,  with VLC, so I'll give her that for this upcoming trip and address all the remote connection issues when we get back.

**Reply by f520k510** [reply]
https://apps.apple.com/hk/app/oplayer-video-player/id344784375?l=en

Here is the link to oplayer on the hk app store
I am not sure if it is available in your region's app store.
